### PR TITLE
Fix validation of literal objects containing no properties

### LINKED ISSUE #3370: NonNullGraphType of input not respected when input is provided inline
Hi, 
In the following code, TestInputType class has a non-null field called prop1.
If the input is provided by variable (result2) , it will get validated correctly if prop1 is not provided, however, when input is provided inline(result1) in the query no validation error will be thrown.

```csharp

async Task Main()
{
	var schema = new Schema { Query = new TestQuery() };
	
	// THIS DOES NOT THROW VALIDATION ERROR
	var result1 = await schema.ExecuteAsync(options =>
			{
				options.Query = @"{test(input: {}) { stringValue }}";
			}
	);
	result1.Dump("Result 1");
	
	// THIS DOES THROW VALIDATION ERROR
	var result2 = await schema.ExecuteAsync(options =>
			{
				options.Query = @"query($input: TestInputType!){test(input: $input) { stringValue }}";
				options.Variables = "{ input: {}}".ToInputs();
			}
	);	
	result2.Dump("Result 2");
}

public static class StringExtensions
{
	private static readonly GraphQLSerializer _serializer = new GraphQLSerializer();

	public static GraphQL.Inputs ToInputs(this string json)
		=> json == null ? GraphQL.Inputs.Empty : _serializer.Deserialize<GraphQL.Inputs>(json) ?? GraphQL.Inputs.Empty;
}

public class TestQuery : ObjectGraphType
{
	public TestQuery()
	{
		Name = "testQuery";
		Description = "Test description";

		Field<TestResponseType>("test")
			.Arguments(new QueryArgument<NonNullGraphType<TestInputType>> { Name = "input" })
			.Resolve(context =>
			{
				return new TestResponse();
			}
		);
	}
}

public class TestResponseType : ObjectGraphType<TestResponse>
{
	public TestResponseType()
	{
		Field<StringGraphType>("stringValue");
	}
}

public class TestResponse
{
	public string StringValue { get; set; } = "Test";
}

public class TestInputType : InputObjectGraphType<TestInput>
{
	public TestInputType()
	{
		Field<NonNullGraphType<StringGraphType>>("prop1");
	}
}

public class TestInput
{
	public string Prop1 { get; set; }
}

```

### PR DESCRIPTION
**Pull request by Shane32**
Fixes #3370

### CODE REVIEW

File: src/GraphQL/Validation/ValidationContext.cs [hunk 1]
```diff
@@ -470,25 +470,26 @@ public async ValueTask<Variables> GetVariableValuesAsync(IVariableVisitor? visit
 
                 var fields = inputType.Fields.List;
                 var fieldAsts = objValue.Fields;
-                if (fieldAsts == null)
-                    return null;
```
Comment (Shane32) [inline src/GraphQL/Validation/ValidationContext.cs:474]: These lines incorrectly skipped validation of object fields with no properties.

**Comment by Shane32** [comment]
I'll merge today unless there are comments

**Comment by sungam3r** [comment]
Hi. Give me 4-5 hours to check this one and one in parser project. I was still dealing with emails last days.

File: src/GraphQL/Validation/ValidationContext.cs (hunk 1 above)
Comment (sungam3r) [reply src/GraphQL/Validation/ValidationContext.cs:474]: 😮   🤦‍♂️

File: src/GraphQL.Tests/Bugs/Issue3370.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    public async Task ValidatesVariableObjectsCorrectly()
+    {
+        var schema = new Schema { Query = new TestQuery() };
+
+        var result2 = await schema.ExecuteAsync(options =>
+        {
+            options.Query = @"query($input: TestInputType!){test(input: $input) { stringValue }}";
+            options.Variables = @"{ ""input"": {}}".ToInputs();
+        }).ConfigureAwait(false);
+        result2.ShouldBeCrossPlatJson("{\"errors\":[{\"message\":\"Variable \\u0027$input.prop1\\u0027 is invalid. No value provided for a non-null variable.\",\"locations\":[{\"line\":1,\"column\":7}],\"extensions\":{\"code\":\"INVALID_VALUE\",\"codes\":[\"INVALID_VALUE\"],\"number\":\"5.8\"}}]}");
```
Comment (sungam3r) [inline src/GraphQL.Tests/Bugs/Issue3370.cs:26]: @Shane32 I think that we can try soon new C# 11 feature - string literals - for such cases. I did not read much about it yet but I have basic knowledge. It works like GraphQL block strings :)

**Review by sungam3r — APPROVED** [review]
🤔 Backport to v5?

**Comment by Shane32** [comment]
> 🤔 Backport to v5?

I agree.

Speaking of backporting fixes, or lack thereof, be sure not to use Microsoft.Data.SqlClient 5.0.1 unless you enjoy tracking down server crashes and memory leaks - see issue dotnet/SqlClient issue 1810

**Comment by sungam3r** [comment]
I read all comments from aforementioned issue. Good or bad - my company uses Oracle databases 😕 .

**Pull request merged**